### ISSUE TITLE
kill the broker or ack takes a long time, which will cause the cluster traffic to drop instantaneously

### ISSUE THREAD

**Issue by keyboardbobo** [apache/pulsar#13523]
**Is your enhancement request related to a problem? Please describe.**
I have nine brokers，when I kill one, the traffic of the remaining eight brokers will drop within 30 seconds, reach a peak after 30 seconds, and then return to normal. I found that the sending thread is blocked: 
`
    private boolean canEnqueueRequest(SendCallback callback, long sequenceId, int payloadSize) {
        try {
            if (conf.isBlockIfQueueFull()) {
                if (semaphore.isPresent()) {
                    semaphore.get().acquire();
                }
..
`
In other words, if the Producer queue corresponding to some partitions is full, the sending thread will be blocked, affecting the sending of other partitions

**Describe the solution you'd like**
Isolate partitions to avoid sending each other

**Describe alternatives you've considered**
PartitionedProducerImpl can automatically exclude Producers whose queues are full

**Comment by gaozhangmin** [comment]
Let me take a look at this.

**Comment by hangc0276** [comment]
It's an interesting issue. I will keep track it.

**Comment by gaozhangmin** [comment]
@hangc0276  @keyboardbobo  This issue is thrown https://github.com/apache/pulsar/wiki/PIP-74%3A-Pulsar-client-memory-limits. It tried to control the whole client memory usage. not per partition. I think it's the expected behavior.

**Comment by keyboardbobo** [comment]
```
        ProducerBuilder builder = client.newProducer(Schema.STRING)
                .topic("persistent://...")
                .sendTimeout(120, TimeUnit.SECONDS)
                .compressionType(CompressionType.LZ4)
                .enableBatching(true)
                .batchingMaxPublishDelay(10, TimeUnit.MILLISECONDS)
                .batchingMaxMessages(1000)
                .blockIfQueueFull(true);
```
@gaozhangmin The client configuration is as above, and the memoryLimitBytes parameter is not set. The main problem now is that a single broker has a problem, which will cause the traffic of other brokers to also drop. It is not that I want to limit the client memory or queue size.

**Comment by gaozhangmin** [comment]
Recently, I upgrade broker, also found this problem, when kill on broker, other broker's bundle traffic will drop off

**Comment by keyboardbobo** [comment]
Theoretically speaking, only a slow response from any broker or a slow network will drag down other producers, because the slowest producer will always be the bottleneck

**Comment by keyboardbobo** [comment]
By isolating the problematic producer, I can prevent the cluster traffic from dropping

**Comment by keyboardbobo** [comment]
I modified the source code, and by isolating the problematic producer, I can successfully prevent the cluster traffic from dropping.

**Comment by gaozhangmin** [comment]
@keyboardbobo  You can  commit your source code if you want.
